### ISSUE TITLE
Invalid report for \Serializable::serialize

### ISSUE THREAD

**Issue by marcospassos** [slevomat/coding-standard#551]
The `TypeHintDeclaration` sniff incorrectly recommends adding a type-hint to the `$serialized` parameter of `\Serializable::serialize`.

Warning: `Method Foo::unserialize() does not have a parameter type hint nor @param annotation...`
```php
class Foo implements \Serializable {
    public function unserialize($serialized) : void {}
}
```
Warning: `Method Foo::unserialize() does not have a parameter type hint for its parameter $serialized but it should be possible to add it...` 
```php
class Foo implements \Serializable {
 
    /**
     * @param string $serialized
     */
    public function unserialize($serialized) : void {}
}
```

**Comment by kukulich** [comment]
You can use @phpcsSuppress annotation.

**Comment by marcospassos** [comment]
Sorry, but why did you close the issue? It is a bug since I can't impose a more strict the parameter type in this case.

**Comment by marcospassos** [comment]
@kukulich could you please reopen this issue?

**Comment by kukulich** [comment]
@marcospassos No, I explained everything in my previous comment.
